addons/sourcemod/scripting/nd_discord_playerlist.sp

```diff
@@ -0,0 +1,193 @@
+    SQL_TQuery(DatabaseHandle, Setup_Step1_CheckTable, query);
+}
+
+void Setup_Step1_CheckTable(Handle owner, Handle handle, const char[] error, any:data)
+{
+    // Create the table if it doesn't exist
+    if (error[0] != '\0')
+    {
+        char query[1024];
+        FormatEx(query, sizeof(query), "CREATE TABLE `players` (`id` bigint NOT NULL AUTO_INCREMENT, `name` varchar(128) NOT NULL, `team` varchar(128) NOT NULL, PRIMARY KEY (`id`));");
+        SQL_TQuery(DatabaseHandle, Setup_Step2_CreateTable, query);
+    }
+}
+
+void Setup_Step2_CreateTable(Handle owner, Handle handle, const char[] error, any:data)
+{
+    if (error[0] != '\0')
+    {
+        SetFailState("Unable to create \"players\" table in \"discord\" database: %s", error);
+    }
+}
+
+void SaveTeams()
+{
+    delete SaveTeamsTimer;
+    SaveTeamsTimer = CreateTimer(1.0, Timer_SaveTeams, _, TIMER_FLAG_NO_MAPCHANGE);
+}
+
+public Action Timer_SaveTeams(Handle timer)
+{
+    if(SavingTeams) {
+        SaveTeamsTimer = CreateTimer(1.0, Timer_SaveTeams, _, TIMER_FLAG_NO_MAPCHANGE);
+        return Plugin_Continue;
+    }
+    SaveTeamsTimer = null;
+
+    if(!IsValidHandle(DatabaseHandle)) {
+        LogAction(0, -1, "Database not connected");
+        return Plugin_Continue;
+    }
+
+    SavingTeams = true;
+
+    char query[1024];
+    FormatEx(query, sizeof(query), "TRUNCATE TABLE players;");
+    SQL_TQuery(DatabaseHandle, SaveTeams_Step1_RemovePlayers, query);
+
+    return Plugin_Continue;
+}
+
+void SaveTeams_Step1_RemovePlayers(Handle owner, Handle handle, const char[] error, any:data)
+{
+    if (error[0] != '\0')
+    {
+        LogAction(0, -1, "Removing players failed: %s", error);
+        SavingTeams = false;
+        return;
+    }
+
+    int[] consortium = new int[MAXPLAYERS + 1];
+    int numConsortium = 0;
+    int[] empire = new int[MAXPLAYERS + 1];
+    int numEmpire = 0;
+    int[] spectator = new int[MAXPLAYERS + 1];
+    int numSpectator = 0;
+    for (int client = 1; client <= MaxClients; client++)
+    {
+        if (IsValidClient(client, false))
+        {
+            int team = GetClientTeam(client);
+            if (team == TEAM_CONSORT)
+            {
+                consortium[numConsortium] = client;
+                numConsortium++;
+            }
+            else if(team == TEAM_EMPIRE)
+            {
+                empire[numEmpire] = client;
+                numEmpire++;
+            }
+            else
+            {
+                spectator[numSpectator] = client;
+                numSpectator++;
+            }
+        }
+    }
+
+    // We're done if we have no players to add
+    if(numConsortium + numEmpire + numSpectator == 0) {
+        SavingTeams = false;
+        return;
+    }
+
+    char query[10000];
+    char queryvalues[10000];
+    char[][] queryparts = new char[numConsortium + numEmpire + numSpectator][1024];
+    int part = 0;
+    char escapedName[MAX_NAME_LENGTH];
```

Comment:
When defining the maximum length for an escaped string you would use the formula [2 * <normal maximum length> + 1].  Right now if you have a client who is already at the maximum name length and has any characters which need escaping then this will not display correctly.

Reply:
done